lib/plugin.js

```diff
@@ -94,20 +94,16 @@ ManifestPlugin.prototype.apply = function(compiler) {
 
     var json = JSON.stringify(cache, null, 2);
 
-    compilation.assets[outputName] = {
-      source: function() {
-        return json;
-      },
-      size: function() {
-        return json.length;
-      }
-    };
-
-    if (this.opts.writeToFileEmit) {
-      var outputFolder = compilation.options.output.path;
-      var outputFile = path.join(outputFolder, this.opts.fileName);
+    var outputFolder = compilation.options.output.path;
+    var outputFile = path.join(outputFolder, this.opts.fileName);
 
+    const shouldUseNodeFs = compiler.outputFileSystem.constructor.name !== 'MemoryFileSystem' ||
+                                      this.opts.writeToFileEmit;
+    if (shouldUseNodeFs) {
```

Comment:
As far as I see, the logic used here breaks what was achieved with https://github.com/danethurber/webpack-manifest-plugin/pull/28, as it assumes the file should be output to both memory and file system - now when `writeToFileEmit` is set ti true, it will be output only to file system, and not to memory.